packages/cozy-client/src/models/file.spec.js

```diff
@@ -29,4 +29,36 @@ describe('File Model', () => {
     }
     expect(file.isNote(note2)).toBe(true)
   })
+
+  describe('normalizeFile', () => {
+    const id = 'uuid123'
+    const type = 'io.cozy.files'
+
+    it('should allow both `id` and `_id`', () => {
+      const doc = { id, type }
+      expect(file.normalize(doc)._id).toBe(id)
+
+      const _doc = { _id: id, type }
+      expect(file.normalize(_doc)._id).toBe(id)
+    })
+
+    it('should allow both `type` and `_type`', () => {
+      const doc = { id, type }
+      expect(file.normalize(doc)._type).toBe(type)
+
+      const _doc = { id, _type: type }
+      expect(file.normalize(_doc)._type).toBe(type)
+    })
+
+    it('should default to "io.cozy.files"', () => {
+      const doc = { id }
+      expect(file.normalize(doc)._type).toBe('io.cozy.files')
+    })
+
+    it('should define the path if parent is provided', () => {
+      const doc = { id, name: 'myfile.ext' }
+      const parent = { path: '/my/path' }
+      expect(file.normalize(doc, parent).path).toBe('/my/path/myfile.ext')
+    })
+  })
```

Comment:
Can you add a test when there is no path? Since you have a condition on it

Reply:
A parent is a directory, so it should always have a path. I may throw an error if a path is missing but I usualy test only how a function should be used, and not what erroneous parameters could do.

Reply:
I have added two tests, one on missing `parent` and one for a `parent` without `path`. Is it what you missed ?